jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ProxyConnectionFactory.java

```diff
@@ -380,6 +378,23 @@ private enum Transport
             0x0D, 0x0A, 0x0D, 0x0A, 0x00, 0x0D, 0x0A, 0x51, 0x55, 0x49, 0x54, 0x0A
         };
 
+        private static final int PP2_TYPE_ALPN = 0x01;
+        private static final int PP2_TYPE_AUTHORITY = 0x02;
+        private static final int PP2_TYPE_CRC32C = 0x03;
+        private static final int PP2_TYPE_NOOP = 0x04;
+        private static final int PP2_TYPE_UNIQUE_ID = 0x05;
+        private static final int PP2_TYPE_SSL = 0x20;
+        private static final int PP2_SUBTYPE_SSL_VERSION = 0x21;
+        private static final int PP2_SUBTYPE_SSL_CN = 0x22;
+        private static final int PP2_SUBTYPE_SSL_CIPHER = 0x23;
+        private static final int PP2_SUBTYPE_SSL_SIG_ALG = 0x24;
+        private static final int PP2_SUBTYPE_SSL_KEY_ALG = 0x25;
+        private static final int PP2_TYPE_NETNS = 0x30;
+
+        private static final int PP2_CLIENT_SSL = 0x01;
+        private static final int PP2_CLIENT_CERT_CONN = 0x02;
+        private static final int PP2_CLIENT_CERT_SESS = 0x04;
+
```

Comment:
There was a bug in the previous code of using `PP2_TYPE_ALPN` rather than `PP2_CLIENT_SSL`, I think because all the values from the spec were not listed.
Having the constants exactly as they are from the spec has value and makes maintenance and understanding simpler.     It is feature that we can see which values are not implemented!